src/libsyntax/tokenstream.rs

```diff
@@ -225,13 +234,107 @@ impl TokenStream {
         }
         true
     }
+
+    pub fn as_tree(self) -> (TokenTree, bool /* joint? */) {
```

Comment:
Also, since this takes `self` by value, should this be named `into_tree` instead? Or are the conventions different for reference counted things